### PR TITLE
Initial add of SNAX DIMC

### PR DESCRIPTION
**Pull request by rgantonio**
This PR adds the DIMC of @mace1997.
It is the initial build where it connects directly to SNAX.

Major TODOs:
- [x] Add wrapper
- [x] Add config for SNAX dimc
- [x] Import SNAX dimc to bender
- [x] Test build
- [x] Add build to CI

### CODE REVIEW

**Review by xiaoling-yi — APPROVED** [review]

File: Bender.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ dependencies:
   snax-gemm:          { git: "https://github.com/KULeuven-MICAS/snax-gemm", rev: main }
   snax-postprocessing-simd: { git: "https://github.com/KULeuven-MICAS/snax-postprocessing-simd", rev: main }
   hypercorex:         { git: "https://github.com/KULeuven-MICAS/hypercorex.git", rev: main }
+  dimc:               { git: "https://github.com/mace1997/dimc.git", rev: main }
```
Comment (JosseVanDelm) [inline Bender.yml:36]: Can we please include this from KULeuven-MICAS instead?
```suggestion
  dimc:               { git: "https://github.com/KULeuven-MICAS/dimc.git", rev: main }
```

Comment (rgantonio) [reply Bender.yml:36]: Oh yeah, wait, we'll fix it on Mace's side.

Comment (JosseVanDelm) [reply Bender.yml]: :pray:

**Comment by rgantonio** [comment]
Merging! It's fixed now. Good job @mace1997 !!!

**Pull request merged**